### ISSUE TITLE
WPS345 and custom types

### ISSUE THREAD

**Issue by vnmabus** [wemake-services/wemake-python-styleguide#1924]
### What's wrong

WPS345 should not be raised when custom types are involved. Code such as
```python
my_custom_obj * 0
```
cannot be replaced by 0, and the object may not allow assigning 0 to it directly.

### How it should be

These checks should only work for builtin types, and maybe a user-defined list of custom types (e.g. Numpy arrays).

### Flake8 version and plugins

Irrelevant.

### pip information

Irrelevant.

### OS information

All OS.

**Comment by sobolevn** [comment]
What `my_custom_obj` is exaclty? Why does it behave this way?

**Comment by vnmabus** [comment]
In my particular case, it is a set of discretized functions, but in general it could be an arbitrary element of a ring, I guess. I currently do not have implemented `__setitem__` for this type, nor a function like `np.zeros_like`, so multiplying by zero is a nice way to obtain a zero function with the same discretization points as another. Sure, I could add that functionality, but I am not sure that a style checker should dictate me which methods I must add...

**Comment by sobolevn** [comment]
> Sure, I could add that functionality, but I am not sure that a style checker should dictate me which methods I must add

😆 I like this! 

In case of `wps` - it probably will do this.

The thing about type of `my_custom_obj` is that we cannot really know about it in `flake8`
But, the rule of thumb is not to relly on `* 0` to produce something meaningful.

**Comment by vnmabus** [comment]
Yes, I guess that it is difficult to know the types in a flake8 plugin. I think that I can take a different approach, at least for my types, or simply ignore it with `noqa` as this pattern is not that common.

Feel free to close this issue if you want.

**Comment by sobolevn** [comment]
Thanks a lot for the report! 👍
